x-pack/plugin/relevance-search/src/main/java/org/elasticsearch/xpack/relevancesearch/relevance/settings/RelevanceSettingsService.java

```diff
@@ -101,6 +109,10 @@ public void clusterChanged(ClusterChangedEvent event) {
             // wait for state recovered
             return;
         }
+        if (this.clusterService.state().metadata().hasIndex(ENT_SEARCH_INDEX)) {
+            // no need to re-create
```

Comment:
in this case we might want to add  `this.clusterService.removeListener(this);` too?